plugin-server/src/cdp/services/hog-watcher.service.test.ts

```diff
@@ -353,4 +378,106 @@ describe('HogWatcher', () => {
             ])
         })
     })
+
+    describe('function type cost differences', () => {
```

Comment:
General comment - these tests feel very verbose and copy-paste like. Feels like a bunch of this could have been dome with an `it.each` with a simple comparison of input and expected output.